src/core/vector/qgsvectorlayer.h

```diff
@@ -2893,6 +2905,9 @@ class CORE_EXPORT QgsVectorLayer : public QgsMapLayer, public QgsExpressionConte
     //! The user-defined actions that are accessed from the Identify Results dialog box
     QgsActionManager *mActions = nullptr;
 
+    //! Flag indicating whether the data source is in read-only mode (editing disabled) or not
+    bool mDSReadOnly = false;
```

Comment:
Can you please expand this variable name to something human intelligible and add some more explanation about what is the difference from `mReadOnly` and how they relate?
```suggestion
    bool mDataSourceReadOnly = false;
```